### DISCUSSION TITLE
Tag the very start of every new release branch

### DISCUSSION THREAD

**Discussion by bryevdv** [bokeh/bokeh#13040] [Build / CI]
Currently, until there is a tag on a new release branch, e.g. `branch-3.2` you see dev versions the reference the previous release:
```
Bokeh version       :  3.1.0rc1+49.g67e2eba9
```
I'd like to propose that the start of every new `branch-x.y` release branch be immediately tagged as a matter of policy. We could use something like `3.2.0.dev0`, since we usually start actual dev releases with `dev1`, or maybe there is something else? As long as it sorts before "dev" versions I think anything sensible is OK. With a manual local tag as a test, we get a more accurate and useful version:
```
Bokeh version       :  3.2.0.dev0+10.g67e2eba9
```
It would be nice if we could just automate this as part of release machinery, but I don't think that is feasible since we usually create a new release level branch ahead of the next release. However, we probably could make a GH action for creating new release level branches, instead of doing it manually. A GH could take care of the tagging, making sure the version makes senses, etc. I would be :+1: since it would help anyone be able to correctly create a new release level branch, rather than just a few people who know the correct magic incantations to get it right. 

As an aside: I *also* wonder if doing this would solve the "0.0.1" issue without requiring everyone to fork with full history? I think the setuptools SCM tool just needs *some* tag, and perhaps the new GH default fork would at least preserve tags on the default branch? (unclear, but it would be nice if so)

cc @bokeh/dev

**Comment by mattpap** [comment]
This has been bothering me as well for a few recent releases.

> We could use something like 3.2.0.dev0 (...)

This seems reasonable to me.

**Comment by tcmetzger** [comment]
+1 from me, that sounds great! Automating this with a GH Action would be very helpful, to make sure this always works in a consistent and reliable way!
